include/SFML/Network/Socket.hpp

```diff
@@ -31,13 +31,37 @@
 
 #include <SFML/Network/SocketHandle.hpp>
 
+#include <algorithm>
+#include <array>
+#include <cstdint>
 #include <vector>
 
 
 namespace sf
 {
 class SocketSelector;
 
+////////////////////////////////////////////////////////////
+/// \brief Robust endianness conversion
+/// \see https://mklimenko.github.io/english/2018/08/22/robust-endian-swap/
+///
+////////////////////////////////////////////////////////////
+template <typename T>
+T swapEndian(T& val, typename std::enable_if<std::is_arithmetic<T>::value, std::nullptr_t>::type = nullptr)
+{
+    union U
+    {
+        T                                   val;
+        std::array<std::uint8_t, sizeof(T)> raw;
+    } src, dst;
+
+    src.val = val;
+    std::reverse_copy(src.raw.begin(), src.raw.end(), dst.raw.begin());
```

Comment:
Doesn't this constitute type punning through a union and thus is UB?

Reply:
Switched to not use a union.